components/public-api-server/pkg/idp/idp.go

```diff
@@ -0,0 +1,167 @@
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		cfg := oidc.DiscoveryConfiguration{
+			Issuer:          kp.IssuerBaseURL,
+			ScopesSupported: op.DefaultSupportedScopes,
+			ResponseTypesSupported: []string{
+				string(oidc.ResponseTypeCode),
+				string(oidc.ResponseTypeIDTokenOnly),
+				string(oidc.ResponseTypeIDToken),
+			},
+			GrantTypesSupported: []oidc.GrantType{
+				oidc.GrantTypeCode,
+				oidc.GrantTypeImplicit,
+			},
+			SubjectTypesSupported: []string{"public"},
+			ClaimsSupported: []string{
+				"sub",
+				"aud",
+				"exp",
+				"iat",
+				"iss",
+				"auth_time",
+				"nonce",
+				"acr",
+				"amr",
+				"c_hash",
+				"at_hash",
+				"act",
+				"scopes",
+				"client_id",
+				"azp",
+				"preferred_username",
+				"name",
+				"family_name",
+				"given_name",
+				"locale",
+				"email",
+			},
+			IDTokenSigningAlgValuesSupported:                   []string{"RS256"},
+			RevocationEndpointAuthMethodsSupported:             []oidc.AuthMethod{oidc.AuthMethodNone},
+			IntrospectionEndpointAuthMethodsSupported:          []oidc.AuthMethod{oidc.AuthMethodNone},
+			IntrospectionEndpointAuthSigningAlgValuesSupported: []string{"RS256"},
+			RequestURIParameterSupported:                       false,
+			AuthorizationEndpoint:                              notSupported,
+			TokenEndpoint:                                      notSupported,
+			IntrospectionEndpoint:                              notSupported,
+			UserinfoEndpoint:                                   notSupported,
+			RevocationEndpoint:                                 notSupported,
+			EndSessionEndpoint:                                 notSupported,
+			JwksURI:                                            keysURL,
+		}
+		err = json.NewEncoder(w).Encode(cfg)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}))
+	mux.Handle("/keys", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		keys, err := kp.keys.PublicKeys(r.Context())
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		_, _ = w.Write(keys)
+	}))
+
+	return mux
+}
+
+func (kp *Service) IDToken(ctx context.Context, org string, audience []string, user oidc.UserInfo) (string, error) {
+	if len(audience) == 0 {
+		return "", fmt.Errorf("audience cannot be empty")
+	}
+	if user == nil {
+		return "", fmt.Errorf("user info cannot be nil")
+	}
+
+	claims := oidc.NewIDTokenClaims(kp.IssuerBaseURL, user.GetSubject(), audience, time.Now().Add(60*time.Minute), time.Now(), "", "", nil, audience[0], 0)
+	claims.SetUserinfo(user)
+
+	codeHash, err := oidc.ClaimHash(string(kp.TokenEncryptionCode), jose.RS256)
+	if err != nil {
+		return "", err
+	}
+	claims.SetCodeHash(codeHash)
+
+	signer, err := kp.keys.Signer(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return crypto.Sign(claims, signer)
```

Comment:
Would be good to catch the error here, and return one which won't contain server specific details to avoid leaking info